src/shell-interface.h

```diff
@@ -95,67 +95,33 @@ struct ShellExternalInterface : ModuleInstance::ExternalInterface {
   } memory;
 
   std::unordered_map<Name, std::vector<Literal>> tables;
+  std::map<Name, std::shared_ptr<ModuleInstance>> linkedInstances;
 
-  ShellExternalInterface() : memory() {}
+  ShellExternalInterface(
+    std::map<Name, std::shared_ptr<ModuleInstance>> linkedInstances_ = {})
+    : memory() {
+    linkedInstances.swap(linkedInstances_);
+  }
   virtual ~ShellExternalInterface() = default;
 
   void init(Module& wasm, ModuleInstance& instance) override {
-    memory.resize(wasm.memory.initial * wasm::Memory::kPageSize);
-
+    if (wasm.memory.exists &&
+        (!wasm.memory.imported() || wasm.memory.module == "env")) {
```

Comment:
why is `env` special-cased? if it's just for testing purposes then that seems risky as `env` is the default name in emscripten.

but i'm not sure why this is needed at all here? if it's imported, it should be initialized in the place it is created, not where it is imported.

Reply:
Oh sorry it's a remnant of the old approach. Fixed.